Fix support of raw files

tmp-hallenser opened the pull request.

tmp-hallenser committed:
bugfix

tmp-hallenser committed:
fix formatting

ildyria commented:
Code wise, no complaints.

I will test it tomorrow with uploading a pdf and some text files ! :D

tmp-hallenser commented:
Unfortunately, there some more work to do. So far, I only fixed the extraction of metadata. 

Open is still the creation of jpgs.

ildyria commented:
At least the current code does not break when I upload ... e.g. pdf. :+1:

tmp-hallenser committed:
Create medium, small and thumb also for raw files

tmp-hallenser commented:
Added the missing lines to create medium, small and thumb jpg files in case of raw files.

Front-end needs to be updated to show work correctly.

Currently, it requires exiftool to work since we need the image dimensions.

tmp-hallenser commented:
Just created a PR for php-exif with imagick as an additional adapter. This will allow uploading raw files without exiftool being installed (some minor changes needed).

ildyria committed:
pull Lychee-front to master, recompile front end, update composer.lock

ildyria commented:
Works fine. I don't have actual raw image to test with on this laptop.

ildyria commented:
@tmp-hallenser merge when you feel like it is ready. :+1:

tmp-hallenser committed:
Use imagemagic to extract metadata in case of raw files and exiftool not available

tmp-hallenser committed:
Update composer.lock

tmp-hallenser committed:
Bugfix

tmp-hallenser committed:
Remove unused import

tmp-hallenser commented:
Now it should be ready for testing

tmp-hallenser committed:
Bugfixes

tmp-hallenser committed:
Bugfix

zhanglix committed:
fix livePhotoUrl path (#590)

tmp-hallenser committed:
Update link to exiftool for travis

tmp-hallenser committed:
Fix extraction of exiftool for travis

ildyria commented on setup_exiftool.sh:
```diff
@@ -2,7 +2,7 @@
 
 wget https://exiftool.org/Image-ExifTool-11.99.tar.gz
-tar -zxvf Image-ExifTool-11.77.tar.gz
+tar -zxvf Image-ExifTool-11.99.tar.gz
```
fixed in master :)

ildyria committed:
merge from master

tmp-hallenser committed:
Bugfix

tmp-hallenser committed:
Merge branch 'fix-564' of https://github.com/LycheeOrg/Lychee into fix-564

tmp-hallenser committed:
Merge branch 'master' into fix-564

tmp-hallenser committed:
Update composer.lock and front js files

tmp-hallenser committed:
Merge branch 'master' into fix-564

tmp-hallenser committed:
Update composer.lock

tmp-hallenser committed:
Bugfixes

* Make raw file extensions case insensitive
* Added try-catch for scaling
* Added try-catch for path of xmp file

tmp-hallenser committed:
Added attempt to create video preview at t=0

In case video thumbnail extaction at middle of video fails, we try a second thumbnail at the beginning of the video

tmp-hallenser commented:
The PR is now ready to merge. Any objections? @ildyria

ildyria committed:
recompile Lychee-front [skip ci]
